contract/models/res_partner.py

```diff
@@ -37,9 +36,9 @@ def act_show_contract(self):
                 self.env.context,
                 search_default_recurring_invoices=True,
                 search_default_not_finished=True,
+                search_default_partner_id=self.id,
                 default_partner_id=self.id,
                 default_recurring_invoices=True,
```

Comment:
add default_pricelist_id=self.pricelist_id.id

Reply:
This is for v11. You might want it for v10, isn't it?